### ISSUE TITLE
Incorrect results for C++ backend

### ISSUE THREAD

**Issue by yav** [GaloisInc/daedalus#226]
The following example produces incorrect result when run on something like this (with non-empty input):
```
def F x =
  block
    a = UInt8
    b = [x]

def Main = F (1 : uint 8)
```
It would appear that `a` always gets the value 0, instead of what's in the file

**Comment by yav** [comment]
Something's quite broken because even just:
```
def Main = UInt8
```
apparently returns 0 always

**Comment by yav** [comment]
This was a silly mistake in the sample driver which should be fixed on PR #225
